Library/Homebrew/diagnostic.rb

```diff
@@ -516,6 +516,22 @@ def check_user_path_3
         EOS
       end
 
+      def check_xdg_data_dirs
+        share = "#{HOMEBREW_PREFIX}/share"
+        homebrew_in_xdg_data_dirs = !ENV["XDG_DATA_DIRS"] ||
+          ENV["XDG_DATA_DIRS"].split(File::PATH_SEPARATOR).include?(share) ||
+          ENV["XDG_DATA_DIRS"] == ""
+        unless homebrew_in_xdg_data_dirs
+          <<-EOS.undent
+          Homebrew's share was not found in your XDG_DATA_DIRS but you have
+          this variable set to include other locations.
+          Some programs like `vapigen` may not work correctly.
+          Consider setting the XDG_DATA_DIRS for example like so
+              echo 'export XDG_DATA_DIRS="#{share}:$XDG_DATA_DIRS"' >> #{shell_profile}
```

Comment:
Might be a good idea. I hope the logic is quite clear? Unset XDG_DATA_DIRS or the one containing Homebrew `share` are considered good. Set but lacking Homebrew `share` is bad.